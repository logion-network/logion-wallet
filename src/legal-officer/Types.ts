import { ProtectionRequest } from '@logion/client/dist/RecoveryClient.js';

import { ColorTheme, rgbaToHex } from '../common/ColorTheme';

export interface RecoveryInfo {
    addressToRecover: string,
    recoveryAccount: ProtectionRequest,
    accountToRecover?: ProtectionRequest,
}

export const LIGHT_MODE: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        foreground: '#000000',
        background: '#eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: '#0d6efd',
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    recoveryItems: {
        iconGradient: {
            from: '#e95b5b',
            to: '#6050dc',
        }
    },
    buttons: {
        secondary: {
            background: '#ffffff',
            foreground: '#000000',
        }
    },
    select: {
        background: '#d8e2fd',
        foreground: '#000000',
        menuBackgroundColor: '#d8e2fd',
        selectedOptionBackgroundColor: rgbaToHex('#ffffff', 0.20),
    },
    table: {
        background: '#ffffff',
        foreground: '#000000',
        header: {
            background: '#ffffff',
            foreground: rgbaToHex('#000000', 0.70),
        },
        row: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    tabs: {
        background: '#ffffff',
        foreground: '#000000',
        inactiveTab: {
            background: '#eff3fe',
            foreground: rgbaToHex('#000000', 0.70),
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#ffffff',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
    dialogTable: {
        background: '#ffffff',
        foreground: '#000000',
        header: {
            background: '#ffffff',
            foreground: rgbaToHex('#000000', 0.70),
        },
        row: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
};
