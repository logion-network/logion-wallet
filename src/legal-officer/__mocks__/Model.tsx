import {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
} from './ModelMock';

export {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
};

export function toRecoveryRequestType(value: string | undefined) {
    return value;
}
