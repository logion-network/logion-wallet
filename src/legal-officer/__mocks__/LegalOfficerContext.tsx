import {
    useLegalOfficerContext,
} from "./LegalOfficerContextMock";

export {
    useLegalOfficerContext
}
