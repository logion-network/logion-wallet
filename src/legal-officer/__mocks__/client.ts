import {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
} from "./ClientMock";

export {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
}
