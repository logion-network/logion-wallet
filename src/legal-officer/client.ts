import {
    EditableRequest,
    Signer,
    SignCallback,
    LocRequestState,
    OpenLoc,
    ClosedLoc,
    ClosedCollectionLoc,
    VerifiedThirdParty,
    VoidedLoc,
    VoidedCollectionLoc,
    LogionClient,
    LegalOfficer,
} from "@logion/client";
import {
    UUID,
    addLink as nodeApiAddLink,
    addFile as nodeApiAddFile,
    addMetadata as nodeApiAddMetadata,
    MetadataItem,
    closeLoc as nodeApiCloseLoc,
    voidLoc as nodeApiVoidLoc,
    VoidInfo,
    asString,
} from "@logion/node-api";
import { AnyJson } from "@polkadot/types-codec/types/helpers.js";
import { fetchAllLocsParams } from "../loc/LegalOfficerLocContext";

export async function addLink(params: {
    locState: EditableRequest,
    target: UUID,
    nature: string,
}): Promise<EditableRequest> {
    const { locState, target, nature } = params;
    const axios = buildAxios(locState);
    await axios.post(`/api/loc-request/${ locState.data().id.toString() }/links`, { target: target.toString(), nature })
    return await locState.refresh() as EditableRequest;
}

export async function deleteLink(params: {
    locState: EditableRequest,
    target: UUID,
}): Promise<EditableRequest> {
    const { locState, target } = params;
    const axios = buildAxios(locState);
    await axios.delete(`/api/loc-request/${ locState.data().id.toString() }/links/${target}`);
    return await locState.refresh() as EditableRequest;
}

function buildAxios(locState: LocRequestState) {
    const client = locState.locsState().client;
    return client.buildAxios(getLegalOfficer(client, locState.data().ownerAddress));
}

function getLegalOfficer(client: LogionClient, address: string): LegalOfficer {
    return client.legalOfficers.find(legalOfficer => address === legalOfficer.address)!;
}

export async function publishLink(params: {
    locState: EditableRequest,
    target: UUID,
    nature: string,
    signer: Signer,
    callback: SignCallback,
}): Promise<EditableRequest> {
    const { locState, target, nature, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locId = locState.data().id;
    const submittable = nodeApiAddLink({
        locId,
        api,
        target,
        nature,
    });
    await signer.signAndSend({
        signerId: locState.data().ownerAddress,
        submittable,
        callback
    });

    const axios = buildAxios(locState);
    await axios.put(`/api/loc-request/${ locId.toString() }/links/${ target.toString() }/confirm`);

    return await locState.refresh() as EditableRequest;
}

export async function publishFile(params: {
    locState: EditableRequest,
    hash: string,
    nature: string,
    submitter: string,
    signer: Signer,
    callback: SignCallback,
}): Promise<EditableRequest> {
    const { locState, hash, nature, submitter, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locId = locState.data().id;
    const submittable = nodeApiAddFile({
        locId,
        api,
        hash,
        nature,
        submitter,
    });
    await signer.signAndSend({
        signerId: locState.data().ownerAddress,
        submittable,
        callback
    });

    const axios = buildAxios(locState);
    await axios.put(`/api/loc-request/${ locId.toString() }/files/${ hash }/confirm`);

    return await locState.refresh() as EditableRequest;
}

export async function publishMetadata(params: {
    locState: EditableRequest,
    item: MetadataItem,
    signer: Signer,
    callback: SignCallback,
}): Promise<EditableRequest> {
    const { locState, item, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locId = locState.data().id;
    const submittable = nodeApiAddMetadata({
        locId,
        api,
        item,
    });
    await signer.signAndSend({
        signerId: locState.data().ownerAddress,
        submittable,
        callback
    });

    const axios = buildAxios(locState);
    await axios.put(`/api/loc-request/${ locId.toString() }/metadata/${ encodeURIComponent(item.name) }/confirm`);

    return await locState.refresh() as EditableRequest;
}

export async function closeLoc(params: {
    locState: OpenLoc,
    signer: Signer,
    callback: SignCallback,
}): Promise<ClosedLoc | ClosedCollectionLoc> {
    const { locState, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locData = locState.data();
    const locId = locData.id;
    const seal = locData.seal;
    const submittable = nodeApiCloseLoc({
        locId,
        api,
        seal,
    });
    const signerId = locState.data().ownerAddress;
    await signer.signAndSend({
        signerId,
        submittable,
        callback
    });

    let refreshedLocState;
    if(locState.discarded) {
        refreshedLocState = (await client.locsState(fetchAllLocsParams(getLegalOfficer(client, signerId)))).findById(locId);
    } else {
        refreshedLocState = locState;
    }

    const axios = buildAxios(refreshedLocState);
    await axios.post(`/api/loc-request/${ locId.toString() }/close`);

    return await refreshedLocState.refresh() as ClosedLoc | ClosedCollectionLoc;
}

export interface FullVoidInfo extends VoidInfo {
    reason: string;
}

export async function voidLoc(params: {
    locState: LocRequestState,
    voidInfo: FullVoidInfo,
    signer: Signer,
    callback: SignCallback,
}): Promise<VoidedLoc | VoidedCollectionLoc> {
    const { locState, voidInfo, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locData = locState.data();
    const locId = locData.id;
    const submittable = nodeApiVoidLoc({
        locId,
        api,
        voidInfo,
    });
    await signer.signAndSend({
        signerId: locState.data().ownerAddress,
        submittable,
        callback
    });

    const axios = buildAxios(locState);
    const reason = voidInfo.reason;
    await axios.post(`/api/loc-request/${ locId.toString() }/void`, {
        reason
    });

    return await locState.refresh() as VoidedLoc | VoidedCollectionLoc;
}

export async function setVerifiedThirdParty(params: {
        locState: ClosedLoc,
        isVerifiedThirdParty: boolean;
    }
): Promise<ClosedLoc> {
    const { locState, isVerifiedThirdParty } = params;
    const axios = buildAxios(locState);
    await axios.put(`/api/loc-request/${locState.data().id.toString()}/verified-third-party`, {
        isVerifiedThirdParty
    });
    return await locState.refresh() as ClosedLoc;
}

export async function getVerifiedThirdPartySelections(params: { locState: OpenLoc } ): Promise<VerifiedThirdParty[]> {
    const { locState } = params
    const axios = buildAxios(locState);
    const response = await axios.get("/api/verified-third-parties");
    const allVerifiedThirdParties: VerifiedThirdParty[] = response.data.verifiedThirdParties;

    const selectedParties = locState.data().selectedParties;

    return allVerifiedThirdParties
        .filter(vtp => vtp.address !== locState.data().requesterAddress)
        .map(vtp => selectedParties.find(selectedParty => selectedParty.identityLocId === vtp.identityLocId && selectedParty.selected) ?
            { ...vtp, selected: true } :
            { ...vtp, selected: false }
        )
        .sort((vtp1, vtp2) => vtp1.lastName.localeCompare(vtp2.lastName));
}

export interface SelectPartiesParams {
    locState: OpenLoc;
    partyId: string;
}

export async function selectParties(params: SelectPartiesParams): Promise<void> {
    const { locState, partyId } = params;
    const axios = buildAxios(locState);
    await axios.post(`/api/loc-request/${locState.locId.toString()}/selected-parties`, {
        identityLocId: partyId
    })
}

export async function unselectParties(params: SelectPartiesParams): Promise<void> {
    const { locState, partyId } = params;
    const axios = buildAxios(locState);
    await axios.delete(`/api/loc-request/${ locState.locId.toString() }/selected-parties/${ partyId }`)
}

export async function requestVote(params: {
    locState: ClosedLoc | ClosedCollectionLoc,
    signer: Signer,
    callback: SignCallback,
}): Promise<string> {
    const { locState, signer, callback } = params;

    const client = locState.locsState().client;
    const api = client.nodeApi;
    const locData = locState.data();
    const locId = locData.id;
    const submittable = api.tx.vote.createVoteForAllLegalOfficers(locId.toDecimalString());
    const result = await signer.signAndSend({
        signerId: locState.data().ownerAddress,
        submittable,
        callback
    });
    const voteCreated = result.events.find(event => event.name === "VoteCreated" && event.section === "vote");
    if(!voteCreated) {
        throw new Error("Unable to retrieve vote ID");
    }
    const voteCreatedData = voteCreated.data as AnyJson[];
    return asString(voteCreatedData[0]);
}
