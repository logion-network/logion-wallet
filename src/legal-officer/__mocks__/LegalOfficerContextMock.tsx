import { LocType, IdentityLocType } from "@logion/node-api/dist/Types";
import { AxiosInstance } from "axios";

import { COLOR_THEME } from "src/common/TestData";
import { RequestAndLoc } from "../LegalOfficerContext";
import { LegalOfficerData } from "../LegalOfficerData";

export let pendingTokenizationRequests: any[] | null = null;

export let rejectRequest = () => {

};

export let tokenizationRequestsHistory: any[] | null = null;

export let refreshRequests = jest.fn();

export let pendingProtectionRequests: any[] | null = null;

export let activatedProtectionRequests: any[] | null = null;

export let protectionRequestsHistory: any[] | null = null;

export let pendingRecoveryRequests: any[] | null = null;

export let recoveryRequestsHistory: any[] | null = null;

export function setPendingRequests(requests: any[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectRequest(callback: any) {
    rejectRequest = callback;
}

export function setTokenizationRequestsHistory(requests: any[]) {
    tokenizationRequestsHistory = requests;
}

export function setRefreshRequests(callback: any) {
    refreshRequests = callback;
}

export function setPendingProtectionRequests(requests: any[]) {
    pendingProtectionRequests = requests;
}

export function setActivatedProtectionRequests(requests: any[]) {
    activatedProtectionRequests = requests;
}

export function setProtectionRequestsHistory(requests: any[]) {
    protectionRequestsHistory = requests;
}

export function setPendingRecoveryRequests(requests: any[]) {
    pendingRecoveryRequests = requests;
}

export function setRecoveryRequestsHistory(requests: any[]) {
    recoveryRequestsHistory = requests;
}

export let pendingLocRequests: Record<LocType, any[]> | null = null;
export let rejectedLocRequests: Record<LocType, any[]> | null = null;

export let openedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let closedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let openedIdentityLocs: RequestAndLoc[] | null = null;

export let openedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let closedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let voidIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export function setRejectedLocRequests(requests: any[]) {
    rejectedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedLocRequests(requests: any[]) {
    openedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setPendingLocRequests(requests: any[]) {
    pendingLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setClosedLocRequests(requests: any[]) {
    closedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedIdentityLocs(requests: any[]) {
    openedIdentityLocs = requests;
}

export function setOpenedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    openedIdentityLocsByType = locs;
}

export function setClosedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    closedIdentityLocsByType = locs;
}

export function setVoidedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    voidIdentityLocsByType = locs;
}

let onchainSettings: LegalOfficerData = {};

export function setOnchainSettings(settings: LegalOfficerData) {
    onchainSettings = settings;
}

export let refreshLocs = jest.fn();

export function useLegalOfficerContext() {
    return {
        pendingTokenizationRequests,
        rejectRequest,
        tokenizationRequestsHistory,
        refreshRequests,
        pendingProtectionRequests,
        activatedProtectionRequests,
        protectionRequestsHistory,
        colorTheme: COLOR_THEME,
        openedLocRequests,
        openedIdentityLocs,
        openedIdentityLocsByType,
        closedLocRequests,
        closedIdentityLocsByType,
        pendingLocRequests,
        rejectedLocRequests,
        voidIdentityLocsByType,
        axios: {} as AxiosInstance,
        onchainSettings,
        pendingRecoveryRequests,
        refreshLocs,
    };
}
