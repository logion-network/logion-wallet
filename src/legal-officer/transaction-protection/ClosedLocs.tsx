import { useNavigate } from "react-router-dom";
import { LocType, IdentityLocType } from '@logion/node-api';

import { useLegalOfficerContext } from "../LegalOfficerContext";
import Table, { Cell, EmptyTableMessage, DateTimeCell, ActionCell } from '../../common/Table';
import LocStatusCell from '../../common/LocStatusCell';
import LocIdCell from '../../common/LocIdCell';
import UserIdentityNameCell from '../../common/UserIdentityNameCell';
import Button from "../../common/Button";
import { locDetailsPath } from "../LegalOfficerPaths";
import ButtonGroup from "../../common/ButtonGroup";
import { useResponsiveContext } from '../../common/Responsive';

export interface Props {
    locType: LocType;
    identityLocType?: IdentityLocType;
}

export default function ClosedLocs(props: Props) {
    const { closedLocRequests, closedIdentityLocsByType } = useLegalOfficerContext();
    const navigate = useNavigate();
    const { width } = useResponsiveContext();
    const { locType, identityLocType } = props

    if (closedLocRequests === null || closedIdentityLocsByType === null) {
        return null;
    }

    const requests = locType === 'Identity' ? closedIdentityLocsByType[identityLocType!] : closedLocRequests[locType];

    return (
        <Table
            columns={[
                {
                    "header": "Requester",
                    render: request => <UserIdentityNameCell userIdentity={ request.userIdentity }/>,
                    align: 'left',
                },
                {
                    "header": "Description",
                    render: request => <Cell content={ request.description } overflowing tooltipId='description-tooltip' />,
                    align: 'left',
                },
                {
                    header: "Status",
                    render: request => <LocStatusCell status={ request.status }/>,
                    width: width({
                        onSmallScreen: '120px',
                        otherwise: '140px'
                    }),
                },
                {
                    header: "LOC ID",
                    render: request => <LocIdCell status={ request.status } id={ request.id } />,
                    align: "left",
                },
                {
                    header: "Creation date",
                    render: request => <DateTimeCell dateTime={ request.createdOn || null } />,
                    width: width({
                        onSmallScreen: '100px',
                        otherwise: '200px'
                    }),
                    align: 'center',
                },
                {
                    header: "Closing date",
                    render: request => <DateTimeCell dateTime={ request.closedOn || null } />,
                    width: width({
                        onSmallScreen: '100px',
                        otherwise: '200px'
                    }),
                    align: 'center',
                },
                {
                    header: "Action",
                    render: request =>
                        <ActionCell>
                            <ButtonGroup>
                                <Button onClick={ () => navigate(locDetailsPath(request.id, request.locType)) }>View</Button>
                            </ButtonGroup>
                        </ActionCell>
                    ,
                    width: width({
                        onSmallScreen: '150px',
                        otherwise: '200px'
                    }),
                    align: 'center',
                }
            ] }
            data={ requests.map(requestAndLoc => requestAndLoc.data()) }
            renderEmpty={ () => <EmptyTableMessage>No LOCs</EmptyTableMessage> }
        />
    );
}
