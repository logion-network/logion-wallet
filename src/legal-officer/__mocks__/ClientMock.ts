import { OpenLoc, VerifiedThirdParty } from "@logion/client";

export let closeLoc = jest.fn().mockResolvedValue(undefined);

export function setCloseLocMock(mock: any) {
    closeLoc = mock;
}

export let deleteLink = jest.fn().mockResolvedValue(undefined);

export let voidLoc = jest.fn().mockResolvedValue(undefined);

export function setVoidLocMock(mock: any) {
    voidLoc = mock;
}

let vtpSelections: VerifiedThirdParty[] = [];

export async function getVerifiedThirdPartySelections(params: { locState: OpenLoc } ): Promise<VerifiedThirdParty[]> {
    return Promise.resolve(vtpSelections);
}

export function setVerifiedThirdPartySelections(mock: VerifiedThirdParty[]) {
    vtpSelections = mock;
}

export let requestVote = jest.fn().mockResolvedValue(undefined);

export function setRequestVoteMock(mock: any) {
    requestVote = mock;
}
