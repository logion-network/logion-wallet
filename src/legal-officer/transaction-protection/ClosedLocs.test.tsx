jest.mock('../../common/CommonContext');
jest.mock('../LegalOfficerContext');

import { ClosedLoc, LocData } from "@logion/client";
import { IdentityLocType } from "@logion/node-api/dist/Types";
import { shallowRender, render } from '../../tests';
import ClosedLocs from './ClosedLocs';
import { setClosedLocRequests, setClosedIdentityLocsByType } from "../__mocks__/LegalOfficerContextMock";
import { UUID } from "@logion/node-api";

describe("ClosedLocs", () => {

    it("renders null with no data", () => {
        const tree = shallowRender(<ClosedLocs locType="Transaction" />);
        expect(tree).toMatchSnapshot();
    });
    
    it("renders pending requests", () => {
        setClosedLocRequests([
            {
                data: () => ({
                    id: new UUID("556f4128-4fc3-4fdc-a543-74e6230911c4"),
                    ownerAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
                    requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
                    description: "LOC description",
                    status: "CLOSED"
                } as unknown as LocData)
            } as ClosedLoc
        ]);
        const tree = shallowRender(<ClosedLocs locType="Transaction" />);
        expect(tree).toMatchSnapshot();
    });
    
    it("renders Closed Polkadot Identity LOCs", () => {
        const locs: Record<IdentityLocType, any[]> = {
            "Polkadot": [ { data: () => locData("John", "Doe") } ],
            "Logion": []
        }
        setClosedIdentityLocsByType(locs);
        const tree = render(<ClosedLocs locType="Identity" identityLocType="Polkadot" />);
        expect(tree).toMatchSnapshot();
    });
    
    it("renders Closed Logion Identity LOCs", () => {
        const locs: Record<IdentityLocType, any[]> = {
            "Polkadot": [],
            "Logion": [ { data: () => locData("Scott", "Tiger") } ]
        }
        setClosedIdentityLocsByType(locs);
        const tree = render(<ClosedLocs locType="Identity" identityLocType="Logion" />);
        expect(tree).toMatchSnapshot();
    });    
});

function locData(firstName: string, lastName: string): LocData {
    return {
        id: new UUID("556f4128-4fc3-4fdc-a543-74e6230911c4"),
        ownerAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        description: "LOC description",
        status: "CLOSED",
        userIdentity: {
            firstName,
            lastName,
        }
    } as unknown as LocData
}
