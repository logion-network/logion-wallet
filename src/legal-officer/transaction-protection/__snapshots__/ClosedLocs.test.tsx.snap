// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Closed Logion Identity LOCs 1`] = `
<div
  className="Table constrained-row-height"
>
  <style>
    
            .Table .body .Row .Col.split-after::before {
                background-color: #152665;
            }
            
  </style>
  <div
    className="header"
    style={
      Object {
        "backgroundColor": "#0c163d",
        "color": "#ffffff",
      }
    }
  >
    <div
      className="Row"
    >
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        Requester
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        Description
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "140px",
          }
        }
      >
        Status
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        LOC ID
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Creation date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Closing date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Action
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div>
      <div
        className="Row "
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell"
            style={Object {}}
          >
            Scott Tiger
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell overflowing"
            style={Object {}}
          >
            <span
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              LOC description
            </span>
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "140px",
            }
          }
        >
          <span
            className="StatusCell"
            style={
              Object {
                "borderColor": "#37ad4b",
              }
            }
          >
            <img
              alt="activated"
              height="20px"
              src="/assets/activated.svg"
            />
             
            <span
              style={
                Object {
                  "color": "#37ad4b",
                  "textTransform": undefined,
                }
              }
            >
              Closed
            </span>
          </span>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="Row"
          >
            <div
              className="regular-cell overflowing"
              style={Object {}}
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                113562046143451386440596991937339789764
              </span>
            </div>
            <button
              className="Button CopyPasteButton medium btn btn-primary"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              type="button"
            >
              <img
                alt="copy_paste"
                src="/assets/copy_paste.svg"
              />
            </button>
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="regular-cell"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            -
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="regular-cell"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            -
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="action-cell"
          >
            <div
              className="ButtonGroup"
            >
              <div
                className="btn-group"
                role="group"
              >
                <button
                  className="Button btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  style={Object {}}
                  type="button"
                >
                  View
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders Closed Polkadot Identity LOCs 1`] = `
<div
  className="Table constrained-row-height"
>
  <style>
    
            .Table .body .Row .Col.split-after::before {
                background-color: #152665;
            }
            
  </style>
  <div
    className="header"
    style={
      Object {
        "backgroundColor": "#0c163d",
        "color": "#ffffff",
      }
    }
  >
    <div
      className="Row"
    >
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        Requester
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        Description
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "140px",
          }
        }
      >
        Status
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 740px ) / 3)",
          }
        }
      >
        LOC ID
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Creation date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Closing date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Action
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div>
      <div
        className="Row "
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell"
            style={Object {}}
          >
            John Doe
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell overflowing"
            style={Object {}}
          >
            <span
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              LOC description
            </span>
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "140px",
            }
          }
        >
          <span
            className="StatusCell"
            style={
              Object {
                "borderColor": "#37ad4b",
              }
            }
          >
            <img
              alt="activated"
              height="20px"
              src="/assets/activated.svg"
            />
             
            <span
              style={
                Object {
                  "color": "#37ad4b",
                  "textTransform": undefined,
                }
              }
            >
              Closed
            </span>
          </span>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 740px ) / 3)",
            }
          }
        >
          <div
            className="Row"
          >
            <div
              className="regular-cell overflowing"
              style={Object {}}
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                113562046143451386440596991937339789764
              </span>
            </div>
            <button
              className="Button CopyPasteButton medium btn btn-primary"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              type="button"
            >
              <img
                alt="copy_paste"
                src="/assets/copy_paste.svg"
              />
            </button>
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="regular-cell"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            -
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="regular-cell"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            -
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="action-cell"
          >
            <div
              className="ButtonGroup"
            >
              <div
                className="btn-group"
                role="group"
              >
                <button
                  className="Button btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  style={Object {}}
                  type="button"
                >
                  View
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders null with no data 1`] = `null`;

exports[`Renders pending requests 1`] = `null`;
